api/apps/api/src/modules/blm/values/repositories/project-blm.api.entity.ts

```diff
@@ -0,0 +1,70 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+import { JsonApiAsyncJobMeta } from '@marxan-api/dto/async-job.dto';
+import { BaseServiceResource } from '@marxan-api/types/resource.interface';
+import { TimeUserEntityMetadata } from '@marxan/utils';
+import { Project } from '@marxan-api/modules/projects/project.api.entity';
+
+export const projectBlmResource: BaseServiceResource = {
+  className: 'ProjectBlm',
+  name: {
+    singular: 'project_blm',
+    plural: 'project_blms',
+  },
+  entitiesAllowedAsIncludes: [],
+};
+
+@Entity('project_blms')
+export class ProjectBlm extends TimeUserEntityMetadata {
+  // Should we use PostgreSQL ranges?
+  @ApiProperty({ isArray: true })
+  @Column('decimal', { array: true, default: [0, 0] })
+  range!: number[];
+
+  @ApiProperty({ isArray: true })
+  @Column('decimal', { array: true, default: [] })
+  values!: number[];
+
+  @ApiProperty({ isArray: true })
+  @Column('decimal', { array: true, default: [] })
+  defaults!: number[];
+
+  /**
+   * The project to which this BLM Values belongs.
+   */
+  @ApiProperty()
+  @PrimaryColumn('uuid')
+  @OneToOne(() => Project)
+  id!: string;
+}
+
+export class JSONAPIProjectBlmData {
```

Comment:
We don't need/want the CRUD there I believe - there is no particular reason to extend internal values storage implementation with full CRUD capabilities.
Also, mixing DTO and DB entity should rather be avoided (unless we are really wanting to build fast) - exposing entity (including all methods) to module consumers can lead to serious issues; also this starts to pollute the entities with possibly unrelated/unwanted related entities.

Reply:
also, extending project/scenario entity (BTW docs/requirements say about scenario, not project, but not sure which is valid tbh, @hotzevzl ?) could leverage `extendGetByIdQuery`-like methods of CRUD abstraction to fetch the values directly.